azure_iot_edge/datadog_checks/azure_iot_edge/check.py

```diff
@@ -0,0 +1,35 @@
+# (C) Datadog, Inc. 2020-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+from typing import cast
+
+from datadog_checks.base import AgentCheck, OpenMetricsBaseCheck
+
+from .config import Config
+from .types import Instance
+
+
+class AzureIotEdgeCheck(AgentCheck):
+    __NAMESPACE__ = 'azure_iot_edge'
+
+    def __init__(self, name, init_config, instances):
+        # type: (str, dict, list) -> None
+        super(AzureIotEdgeCheck, self).__init__(name, init_config, instances)
+        self._config = Config(cast(Instance, self.instance), check_namespace=self.__NAMESPACE__)
+        self._edge_hub_check = OpenMetricsBaseCheck(name, init_config, [self._config.edge_hub_instance])
+        self._edge_agent_check = OpenMetricsBaseCheck(name, init_config, [self._config.edge_agent_instance])
```

Comment:
This way the sub checks will need to run in sequential order.

What about having users to provide separate instance configs for the edge_agent and the edge_hub ? How does the configs differ for the two types ?

Reply:
As discussed on Slack - these OpenMetrics checks are lightweight (1 HTTP request + processing), and composing them like this allows to keep the "1 instance = 1 IoT Edge device" mental model, which is much better from a UX perspective than "remember to configure 3 different instances like this with different options" (not mentioning the associated extra code logic in the check to account for this).

Reply:
This is a helpful explanation, and since the approach differs from how we do most other checks it would be useful to have a comment indicating the rationale.

Reply:
Added a comment with the rationale for this in 4aa56608f :+1: